comp/core/agenttelemetry/fx/fx.go

```diff
@@ -7,17 +7,11 @@
 package fx
 
 import (
-	agenttelemetry "github.com/DataDog/datadog-agent/comp/core/agenttelemetry/def"
 	agenttelemetryimpl "github.com/DataDog/datadog-agent/comp/core/agenttelemetry/impl"
 	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
 )
 
 // Module defines the fx options for this component
 func Module() fxutil.Module {
-	return fxutil.Component(
-		fxutil.ProvideComponentConstructor(
-			agenttelemetryimpl.NewComponent,
-		),
-		fxutil.ProvideOptional[agenttelemetry.Component](),
-	)
+	return agenttelemetryimpl.Module()
```

Comment:
Can you explain why you made this change? 
The idea behind `fx` and `impl` is to make `impl` free of fx.